libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/ViewportDataSourceProcessor.kt

```diff
@@ -193,12 +193,34 @@ internal object ViewportDataSourceProcessor {
         return try {
             val currentStepFullPoints = currentStepProgress.stepPoints ?: emptyList()
             // todo bottom slice might not be needed since we always append the user location
-            TurfMisc.lineSliceAlong(
+            val lineSliceCoordinatesForLookaheadDistance = TurfMisc.lineSliceAlong(
                 LineString.fromLngLats(currentStepFullPoints),
                 distanceTraveledOnStepKM,
                 lookaheadDistanceForZoom,
                 TurfConstants.UNIT_KILOMETERS
             ).coordinates()
+            val outputCoordinates: MutableList<Point> = emptyList<Point>().toMutableList()
+            val firstEdgeBearing = TurfMeasurement.bearing(lineSliceCoordinatesForLookaheadDistance[0], lineSliceCoordinatesForLookaheadDistance[1])
+            outputCoordinates.add(lineSliceCoordinatesForLookaheadDistance[0])
+            for (index in lineSliceCoordinatesForLookaheadDistance.indices) {
+                if (index == 0) {
+                    continue
+                }
+                val coord = lineSliceCoordinatesForLookaheadDistance[index-1]?.let {
+                    val thisEdgeBearing = TurfMeasurement.bearing(it, lineSliceCoordinatesForLookaheadDistance[index])
+                    if (abs(shortestRotationDiff(thisEdgeBearing, firstEdgeBearing)) < 100.0) {
```

Comment:
the bearing can be between -360.0 and 360.0, so the `abs` value can give a large value when it should be small.

[I had an issue in replay from calculating a bearing delta in this way](https://github.com/mapbox/mapbox-navigation-android/pull/4338/files#diff-95cd4488581ea9e43526f46294dc3cfbca8970ccbfa848cbe830da75f732d587R68). Add some tests to ensure it will give accurate results.

Reply:
Does the `shortestRotationDiff` and its call to the `wrap` function take care of this issue? I think it does. If not I'll include the fix you added in that link.

Reply:
oh i'm blind and just read abs(thisbearing, otherbearing). But we should add tests and move it to TurfMeasurements as we have a few places doing the same operation in different ways.

Will you throw this issue as a comment so we can find it later https://github.com/mapbox/mapbox-java/issues/1264